garden_ai/mlmodel.py

```diff
@@ -186,129 +183,6 @@ def clear_mlflow_staging_directory():
-            raise
-
-        extraction_dir = download_dir / "unzipped"
-        unzipped_path = str(download_dir / extraction_dir)
-
-        try:
-            with zipfile.ZipFile(zip_filepath, "r") as zip_ref:
-                zip_ref.extractall(unzipped_path)
-        except (FileNotFoundError, zipfile.BadZipFile) as fe:
-            print(f"Failed to unzip model directory from Garden model repository. {fe}")
-            raise
-
-        return unzipped_path
-
-    @staticmethod
-    def get_download_url(full_model_name: str) -> str:
-        model_url_json = os.environ.get("GARDEN_MODELS", None)
-        if not model_url_json:
-            raise KeyError(
-                "GARDEN_MODELS environment variable was not set. Cannot download model."
-            )
-        try:
-            model_url_dict = json.loads(model_url_json)
-            return model_url_dict[full_model_name]
-        except (json.JSONDecodeError, KeyError):
-            print(
-                f"Could not find url for model {full_model_name} in GARDEN_MODELS env var contents {model_url_json}"
-            )
-            raise
-
-    # Duplicated in this class so that _Model is self-contained
-    @staticmethod
-    def clear_mlflow_staging_directory():
-        staging_dir = pathlib.Path(os.path.expanduser("~/.garden")) / "mlflow"
-        path = str(staging_dir)
-        for item in os.listdir(path):
-            item_path = os.path.join(path, item)
-            if os.path.isfile(item_path):
-                os.remove(item_path)
-            elif os.path.isdir(item_path):
-                shutil.rmtree(item_path)
-
-    def _lazy_load_model(self):
-        """download and deserialize the underlying model, if necessary."""
-        if self.model is None:
-            download_url = self.get_download_url(self.full_name)
-            local_model_path = self.download_and_stage(download_url, self.full_name)
-            self.model = load_model(local_model_path, suppress_warnings=True)
-            try:
-                self.clear_mlflow_staging_directory()
-            except Exception as e:
-                print(
-                    f"Could not clean up model staging directory. Check permissions on {self.staging_dir}"
-                )
-                print(str(e))
-                pass
-        return
-
-    def predict(self, data):
-        """Generate model predictions.
-
-        The underlying model will be downloaded if it hasn't already.
-
-        input data is passed directly to the underlying model via its respective
-        ``predict`` method.
-
-        Parameters
-        ----------
-        data : pd.DataFrame | pd.Series | np.ndarray | List[Any] | Dict[str, Any]
-            Input data fed to the model
-
-        Returns
-        --------
-        Results of model prediction
-
-        """
-        self._lazy_load_model()
-        return self.model.predict(data)
-
-
 @lru_cache
```

Comment:
Is this doing anything meaningful for us if the expensive operation is happening in `_Model`?

Reply:
oh yeah you're totally right, actually think that hasn't been doing anything for us for a while now